### PR TITLE
Add fixes to APIM codebase for migration

### PR DESCRIPTION
**Pull request by Lakith-Rambukkanage**
## Purpose
Fixing : https://github.com/wso2/api-manager/issues/748

## Approach
Add system property for migrations and omit non-required executions during migration time.

## Related PRs
https://github.com/wso2-support/carbon-apimgt/pull/5133
https://github.com/wso2-support/carbon-apimgt/pull/5312

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java [hunk 1]
```diff
@@ -104,18 +104,22 @@ public GatewayStartupListener() {
...
+            } catch (CertificateManagementException e) {
+                log.error("Error while Backup Truststore", e);
+            }
+            log.debug("Registering ServerStartupListener for SubscriptionStore for the tenant domain : " + MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            SubscriptionDataHolder.getInstance().registerTenantSubscriptionStore(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            log.debug("Registered ServerStartupListener for SubscriptionStore for the tenant domain : " + MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            SubscriptionDataHolder.getInstance().initializeSubscriptionStore(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            cleanDeployment(CarbonUtils.getCarbonRepository());
+        } else {
+            log.info("Stopped at gateway startup listener completing!");
```
Comment (dushaniw) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java:121]: Shall we make log more descriptive by adding that "running on migration enabled mode"

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java [hunk 2]
```diff
@@ -155,36 +159,40 @@ private void cleanDeployment(String artifactRepositoryPath) {
...
+            jmsTransportHandlerForEventHub
+                    .subscribeForJmsEvents(APIConstants.TopicNames.TOPIC_NOTIFICATION, new GatewayJMSMessageListener());
+            jmsTransportHandlerForEventHub
+                    .subscribeForJmsEvents(APIConstants.TopicNames.TOPIC_THROTTLE_DATA, new JMSMessageListener());
+            jmsTransportHandlerForEventHub.subscribeForJmsEvents(APIConstants.TopicNames.TOPIC_ASYNC_WEBHOOKS_DATA,
+                    new GatewayJMSMessageListener());
+            copyTenantArtifacts();
+            APILoggerManager.getInstance().initializeAPILoggerList();
+        } else {
+            log.info("Stopped at Gateway Startup listener completed!");
```
Comment (dushaniw) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java:194]: lets improve the log here as well

**Review by dushaniw — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java.orig [hunk 3]
```diff
@@ -0,0 +1,2949 @@
+/*
```
Comment (rmsamitha) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java.orig:1]: Is this an auto generated file from git merge tool? Do we need to maintain this? Please remove this if there is not specific need to maintain this.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 4]
```diff
@@ -5432,6 +5438,12 @@ public void deployAPIRevision(String apiId, String apiRevisionId,
             List<APIRevisionDeployment> apiRevisionDeployments, String organization)
             throws APIManagementException {
         APIIdentifier apiIdentifier = APIUtil.getAPIIdentifierFromUUID(apiId);
+        if (organization ==  null) {
```
Comment (rmsamitha) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:5441]: Formatting issue. Additional space.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java.orig (hunk 3 above)
Comment (Lakith-Rambukkanage) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java.orig:1]: Addressed by :https://github.com/wso2/carbon-apimgt/pull/12044

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 4 above)
Comment (Lakith-Rambukkanage) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:5441]: Addressed by :https://github.com/wso2/carbon-apimgt/pull/12044

**Pull request merged**